node-repository/src/main/java/com/yahoo/vespa/hosted/provision/autoscale/Autoscaler.java

```diff
@@ -126,32 +126,31 @@ private boolean similar(double r1, double r2, double threshold) {
      * Returns the smallest allocatable node resources larger than the given node resources,
      * or empty if none available.
      */
-    private Optional<ClusterResourcesWithCost> toAllocatableResources(ClusterResources resources, ClusterSpec cluster) {
+    private Optional<AllocatableClusterResources> toAllocatableResources(ClusterResources resources, ClusterSpec cluster) {
         if (allowsHostSharing(nodeRepository.zone().cloud())) {
             // Return the requested resources, adjusted to be legal or empty if they cannot fit on existing hosts
             NodeResources nodeResources = nodeResourceLimits.enlargeToLegal(resources.nodeResources(), cluster.type());
             for (Flavor flavor : nodeRepository.getAvailableFlavors().getFlavors())
                 if (flavor.resources().satisfies(nodeResources))
-                    return Optional.of(new ClusterResourcesWithCost(resources.with(nodeResources),
-                                                                    costOf(nodeResources) * resources.nodes()));
+                    return Optional.of(new AllocatableClusterResources(resources.with(nodeResources),
+                                                                       nodeResources));
             return Optional.empty();
         }
         else {
             // return the cheapest flavor satisfying the target resources, if any
-            double bestCost = Double.MAX_VALUE;
-            Optional<Flavor> bestFlavor = Optional.empty();
+            Optional<AllocatableClusterResources> best = Optional.empty();
             for (Flavor flavor : nodeRepository.getAvailableFlavors().getFlavors()) {
                 if ( ! flavor.resources().satisfies(resources.nodeResources())) continue;
-                if (bestFlavor.isEmpty() || bestCost > costOf(flavor.resources())) {
-                    bestFlavor = Optional.of(flavor);
-                    bestCost = costOf(flavor);
-                }
+
+                if (flavor.resources().storageType() == NodeResources.StorageType.remote)
+                    flavor = flavor.with(FlavorOverrides.ofDisk(resources.nodeResources().diskGb()));
+                var candidate = new AllocatableClusterResources(resources.with(flavor.resources()),
+                                                                hostResourcesCalculator.availableCapacityOf(flavor.name(), flavor.resources()));
+
+                if (best.isEmpty() || best.get().cost() > costOf(flavor.resources()))
```

Comment:
```suggestion
                if (best.isEmpty() || best.get().cost() > candidate.cost())
```
Previously this would compare cost of advertised resources for the best match against cost of real resources for the candidate.